os-server/src/main/java/app/openschool/usermanagement/UserServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package app.openschool.usermanagement;
+
+import static org.springframework.http.HttpStatus.CREATED;
+
+import app.openschool.usermanagement.api.dto.UserRegistrationDto;
+import app.openschool.usermanagement.api.dto.UserRegistrationHttpResponse;
+import app.openschool.usermanagement.entities.RoleEntity;
+import app.openschool.usermanagement.entities.UserEntity;
+import app.openschool.usermanagement.exceptions.EmailAlreadyExistException;
+import java.util.Locale;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/** Useful Javadoc. */
+@Service
+public class UserServiceImpl implements UserService {
+
+  public static final String EMAIL_ALREADY_EXISTS = "User with this email already exists";
+  private final UserRepository userRepository;
+  private final RoleRepository roleRepository;
+  private final BCryptPasswordEncoder passwordEncoder;
+
+  /** Useful Javadoc. */
+  public UserServiceImpl(
+      UserRepository userRepository,
+      RoleRepository roleRepository,
+      BCryptPasswordEncoder passwordEncoder) {
+    this.userRepository = userRepository;
+    this.roleRepository = roleRepository;
+    this.passwordEncoder = passwordEncoder;
+  }
+
+  @Override
+  @Transactional
```

Comment:
do you really need transactional here ?

Reply:
I guess yes because I've used two repository methods. RoleRepository.findRoleEntityByType() and UserRepository.save(). I think another option could be saving users with predefined role_id.